### PR TITLE
Add a convenience playbook for docker login

### PR DESCRIPTION
**Pull request by ajdecon**
Providing a Docker login functionality will enable some key workflows, including:

- Using Docker Pro or other accounts that provide greater rate limits than anonymous access
- Accessing private containers that are not available anonymously

For test instructions, see `docs/docker-login.md`. Note that these instructions will walk you through how to put the login credentials in an Ansible Vault file for security.

### CODE REVIEW

File: config.example/group_vars/all.yml [hunk 1]
```diff
@@ -168,6 +168,17 @@ bind_paths: []
...
+#   Note: example only! you should put these in an Ansible Vault file for security!
+#
+#docker_login_registries:
+#- registry: docker.io
+#  username: myuser
+#  password: mypassword
+#  email: docker@docker.io
+#- registry: nvcr.io
+#  username: '$oauthtoken'
+#  password: mypassword
```
Comment (dholt) [inline config.example/group_vars/all.yml:182]: Can we move these up near the other docker options?

**Review by dholt — DISMISSED** [review]
A couple minor organizational comments.

Can we move the documentation to a subfolder in docs? Don't care where, but probably not root. Maybe make a `misc` folder and put this doc and the firewall doc in there?

File: config.example/group_vars/all.yml (hunk 1 above)
Comment (ajdecon) [reply config.example/group_vars/all.yml]: @dholt : Made this change and the recommended move of the doc file.

**Review by michael-balint — COMMENTED** [review]
Playbook runs successfully, but nothing gets saved to `~/.docker/config`. Doing `docker pull` on a container in one of the registries yields an error asking for authentication.

```
TASK [docker-login : log into docker registry] **************************************************
changed: [deepops-dgx1q-05] => (item={'registry': 'nvcr.io', 'username': '$oauthtoken', 'password': '<redacted>'})
changed: [deepops-dgx1q-05] => (item={'registry': 'docker.io', 'username': '<redacted>', 'password': '<redacted>'})
```

Also, if possible, is there a way to scrub the ansible logs above so that the passwords aren't being shown in plaintext?

**Comment by ajdecon** [comment]
Added `no_log` flag to `docker_login` task.

Tested and confirmed that `/root/.docker/config.json` is being set in my tests.

**Review by michael-balint — APPROVED** [review]

**Pull request merged**